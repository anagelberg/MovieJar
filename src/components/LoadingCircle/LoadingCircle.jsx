import './LoadingCircle.scss'; // Import the CSS file

const LoadingCircle = () => {
    return (
        <div className='loading'>
            <div className="loading__circle">
                <div className="loading__circle-inner"></div>
            </div>
        </div>

    );
};

export default LoadingCircle;