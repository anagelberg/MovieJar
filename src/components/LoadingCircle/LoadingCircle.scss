@use "../../styles/mixins" as *;
@use "../../styles/variables" as *;

.loading {
  width: 100%;
  height: 40vh;
  @include flex-center;

  @include tablet {
    height: 60vh;
  }

  &__circle {
    @include flex-center;
    width: 2.5rem;
    height: 2.5rem;
    border-radius: 50%;
    animation: spin 1s linear infinite;
    background: linear-gradient(
      to top,
      $secondary-background 65%,
      $primary 100%
    );
  }

  &__circle-inner {
    width: 1.8rem;
    height: 1.8rem;
    border-radius: 50%;
    background-color: $primary-background;
  }
}

@keyframes spin {
  0% {
    transform: rotate(0deg);
  }
  48% {
    transform: rotate(70deg);
  }
  100% {
    transform: rotate(360deg);
  }
}
