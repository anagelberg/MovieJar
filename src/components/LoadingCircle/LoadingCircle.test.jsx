import { render } from "@testing-library/react";
import LoadingCircle from "./LoadingCircle";

test('renders LoadingCircle component', () => {
    const { container } = render(<LoadingCircle />);
    expect(container.firstChild).toHaveClass('loading');
});

test('LoadingCircle component matches snapshot', () => {
    const { container } = render(<LoadingCircle />);
    expect(container.firstChild).toMatchSnapshot();
});