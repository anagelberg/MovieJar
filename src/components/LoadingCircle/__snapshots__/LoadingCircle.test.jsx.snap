// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoadingCircle component matches snapshot 1`] = `
<div
  class="loading"
>
  <div
    class="loading__circle"
  >
    <div
      class="loading__circle-inner"
    />
  </div>
</div>
`;
